src/avram/database.cr

```diff
@@ -190,20 +199,23 @@ abstract class Avram::Database
     end
   end
 
-  private def transactions
-    self.class.transactions
+  private def connections
+    self.class.connections
   end
 
   private def wrap_in_transaction
-    db.transaction do |tx|
-      transactions[Fiber.current.object_id] ||= tx
+    (current_transaction || current_connection).transaction do
       yield
     end
     true
   rescue e : Avram::Rollback
     false
   ensure
-    transactions.delete(Fiber.current.object_id)
+    # TODO: not sure of this
+    if current_connection.stack.empty?
+      current_connection.release
+      connections.delete(current_object_id)
+    end
```

Comment:
I was referencing Rails' implementation for all of this. They had the tie between the connection and the thread id in the database pool which made this a bit confusing. Not sure if this is correct.

Reply:
When would the stack on the current_connection be empty? Like, what would cause that case?

Reply:
Outside of specs where a root transaction is created, once an operation is complete, the transaction is done and the stack will be empty so we release the connection back to the thread pool. This is where things are pretty confusing to me at the moment